prime-router/src/main/kotlin/tokens/TokenAuthentication.kt

```diff
@@ -0,0 +1,220 @@
+            lookup: ReportStreamSecretFinder,
+            actionHistory: ActionHistory? = null,
+        ): AccessToken {
+            if (scopeAuthorized.isEmpty() || scopeAuthorized.isBlank()) error("Empty or blank scope request")
+            val secret = lookup.getReportStreamTokenSigningSecret()
+            // Using Integer seconds to stay consistent with the JWT token spec, which uses seconds.
+            // Search for 'NumericDate' in https://tools.ietf.org/html/rfc7519#section-2
+            // This will break in 2038, which, actually, isn't that far off...
+            val expiresInSeconds = 300
+            val expiresAtSeconds = ((System.currentTimeMillis()/1000) + expiresInSeconds).toInt()
+            val expirationDate = Date(expiresAtSeconds.toLong() * 1000)
+            // Keep it small:  The signed token we send back only has two claims in it.
+            val token = Jwts.builder()
+                .setExpiration(expirationDate)  // exp
+                .claim("scope", scopeAuthorized)
+                .signWith(secret).compact()
+            actionHistory?.trackActionResult(
+                "Token successfully created for $scopeAuthorized. Expires at $expirationDate")
+            return AccessToken(token, "bearer", expiresInSeconds, expiresAtSeconds, scopeAuthorized)
+        }
+
+        fun checkAccessToken(request: HttpRequestMessage<String?>, desiredScope: String): Claims? {
+            val bearerComponent = request.headers["authorization"]
+            if (bearerComponent == null) {
+                logger.error("Missing Authorization header.  Unauthorized.")
+                return null
+            }
+            val accessToken = bearerComponent.split(" ")[1]
+            if (accessToken.isEmpty()) {
+                logger.error("Request has Authorization header but no token.  Unauthorized")
+                return null
+            }
+// for testing only           return checkAccessToken(accessToken, desiredScope, GetStaticSecret())
+            return checkAccessToken(accessToken, desiredScope, FindReportStreamSecretInVault())
+        }
+
+        /**
+         * lookup is a call back to get the ReportStream secret used to sign the token.  Using a callback
+         * makes this easy to test - can pass in a static test secret
+         * This does not need to be a public/private key.
+         */
+        fun checkAccessToken(accessToken: String, desiredScope: String, lookup: ReportStreamSecretFinder): Claims? {
+            try {
+                val secret = lookup.getReportStreamTokenSigningSecret()
+                // Check the signature.  Throws JwtException on problems.
+                val jws = Jwts.parserBuilder()
+                    .setSigningKey(secret)
+                    .build()
+                    .parseClaimsJws(accessToken)
+                if (jws.body == null) {
+                    logger.error("AccessToken check failed - no claims.  Unauthorized.")
```

Comment:
See my comment about creating a unique id for each access token to track in logs.

Reply:
Done.